src/tess.py

```diff
@@ -0,0 +1,364 @@
+            for filt in filters:
+                df = df_gap[df_gap[filt_col] == filt].reset_index(
+                    drop=True, inplace=False
+                )
+
+                # drop the flagged or NaN values
+                if drop_bad:
+                    bad_idx = (df[flag_col] != 0) | (df[mag_col].isna())
+                    df = df[~bad_idx]
+
+                if len(df) > 0:
+                    dfs.append(df)
+
+            prev_idx = idx + 1
+
+        new_datasets = []
+        keep_columns = [
+            time_col,
+            exp_col,
+            filt_col,
+            mag_col,
+            magerr_col,
+            ra_col,
+            dec_col,
+            flag_col,
+        ]
+        for df in dfs:
+            df = df[keep_columns]
+            if len(df) > 0:
+                new_datasets.append(Lightcurve(data=df, **init_kwargs))
+
+        return new_datasets
+
+    def fetch_data_from_observatory(self, cat_row, verbose=0, **kwargs):
+        """
+        Fetch data from TESS for a given source.
+        Must return a dataframe (or equivalent), even if it is an empty one.
+        This must be implemented by each observatory subclass.
+
+        Parameters
+        ----------
+        cat_row: dict like
+            A row in the catalog for a specific source.
+            In general, this row should contain the following keys:
+            name, ra, dec, mag, filter_name (saying which band the mag is in).
+        kwargs: dict
+            Additional keyword arguments to pass to the fetcher.
+
+        Returns
+        -------
+        data : pandas.DataFrame or other data structure
+            Raw data from the observatory, to be put into a RawData object.
+        altdata: dict
+            Additional data to be stored in the RawData object.
+
+        """
+        # TODO: figure out dstArcSec threshold for flagging close stars
+        DST_THRESHOLD = 10
+        # TODO: figure out mag diff threshold for flagging similar stars
+        MAGDIFF_THRESHOLD = 0.5
+        # TODO: figure out optimal radii for queries
+        
+        # name should be source_id for GAIA catalog entry
+        name = cat_row['name']
+        dec = cat_row['dec']
+        ra = cat_row['ra']
+        mag = cat_row['mag']
+        if mag > 16:
+            # TESS can't see stars fainter than 15 mag
+            print(f"Magnitude of {mag} is too faint for TESS.")
+            return pd.DataFrame()
+
+        cone_radius = kwargs['cone_radius'] \
+                        if 'cone_radius' in kwargs else 0.01
+        sign = ' -' if dec < 0 else ' '
+        coord_str = str(ra) + sign + str(dec)
+        
+        kwargs = {
+            'coordinates': coord_str,
+            'catalog': 'TIC',
+            'radius': cone_radius
+        }
+        catalog_data = self._try_query(Catalogs.query_region, kwargs)
+        if len(catalog_data) == 0:
+            print("No TESS object found for given catalog row.")
```

Comment:
add `verbose` check. The goal is to have everything running totally quietly (unless there's an exception that we can't handle), and only print things out when verbose is non-zero (we could also make it a numeric value, e.g., only print if `verbose>2` but that seems too complicated for now).